pkg/apis/scorecard/v1alpha1/types.go

```diff
@@ -0,0 +1,121 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// State is a type used to indicate the result state of a Test.
+type State string
+
+const (
+	// UnsetState is the default state for a TestResult. It must be updated by UpdateState or by the Test.
+	UnsetState State = "unset"
+	// PassState occurs when a Test's ExpectedPoints == MaximumPoints.
+	PassState State = "pass"
+	// PartialPassState occurs when a Test's ExpectedPoints < MaximumPoints and ExpectedPoints > 0.
+	PartialPassState State = "partial_pass"
+	// FailState occurs when a Test's ExpectedPoints == 0.
+	FailState State = "fail"
+	// ErrorState occurs when a Test encounters a fatal error and the reported points should not be considered.
+	ErrorState State = "error"
+)
+
+// ScorecardTestSpec contains info about a the test in a ScorecardTest object. In the future, this will likely
+// be updated to contain more info about how the tests can be run
+// +k8s:openapi-gen=true
+type ScorecardTestSpec struct {
+	// Name is the name of the test, which should be obtained from TestInfo
+	Name string `json:"name"`
+	// Description is the description of the test, which should be obtained from TestInfo
+	Description string `json:"description"`
+}
+
+// ScorecardTestResult stores the results of a scorecard tests and the logs from the tests
+// +k8s:openapi-gen=true
+type ScorecardTestResult struct {
+	// Log contains the scorecard's current log.
+	Log string `json:"log"`
+	// Results is an array of ScorecardResult for each suite of the curent scorecard run.
+	Results []ScorecardResult `json:"results"`
+}
+
+// ScorecardResult contains the combined results of a suite of tests
+// +k8s:openapi-gen=true
+type ScorecardResult struct {
+	// Error is the number of tests that ended in the Error state
+	Error int `json:"error"`
+	// Pass is the number of tests that ended in the Pass state
+	Pass int `json:"pass"`
+	// PartialPass is the number of tests that ended in the PartialPass state
+	PartialPass int `json:"partialPass"`
+	// Fail is the number of tests that ended in the Fail state
+	Fail int `json:"fail"`
+	// TotalTests is the total number of tests run in this suite
+	TotalTests int `json:"totalTests"`
+	// TotalScore is the total score of this quite as a percentage
+	TotalScore int `json:"totalScorePercent"`
+	// Tests is an array containing a json-ified version of the TestResults for the suite
+	Tests []*JSONTestResult `json:"tests"`
+}
+
+// JSONTestResult is a simplified version of the TestResult that only include the Name and Description of the TestInfo field in TestResult
+// +k8s:openapi-gen=true
+type JSONTestResult struct {
```

Comment:
This could be output as yaml as easily as JSON, Don't think there is anything in the struct to necessitate the `JSON` prefix.